File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2022s/IZ08IO3/Strategy/src/Sportsmen.java
public class Sportsmen
{
   private String surname;
   private SportsmenStrategyRun sRun;
   public Sportsmen(String surname,SportsmenStrategyRun sRun)
   {
      this.surname=surname;
      this.sRun=sRun;
   }
   public String getSurname() { return surname; }

   public String jump() { return "Jumping"; }
   public String run() { return sRun.run(); }
   public String swim() { return "Swimming"; }
   String execute(SportsmenCommand cmd) { return surname+": "+cmd.execute(this); }
}

File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2022s/ID06PD1/StrategyWkitoria/Strategy/Command/Sportsmen.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Strategy
{

    internal class Sportsmen
    {
        public string surname { get; private set; }
        private SportsmenStrategyRun stRun;
        private SportsmenStrategyJump stJump;
        private SportsmenStrategySwim stSwim;

        public Sportsmen(string surname, SportsmenStrategyRun stRun, SportsmenStrategyJump stJump, SportsmenStrategySwim stSwim)
        {
            this.surname = surname;
            this.stRun = stRun;
            this.stJump = stJump;
            this.stSwim = stSwim;
        }

        public string Run() { return stRun.ExecuteRun(); }
        public string Jump() { return stJump.ExecuteJump(); }
        public string Swim() { return stSwim.ExecuteSwim(); }   
        
        public string execute(SportsmenCommand command)
        {
            return surname + ": " + command.execute(this);
        }
    }
}
